lldb/include/lldb/Core/ValueObject.h

```diff
@@ -441,6 +441,19 @@ class ValueObject {
 
   virtual int64_t GetValueAsSigned(int64_t fail_value, bool *success = nullptr);
 
+  llvm::APSInt GetValueAsAPSInt();
```

Comment:
Not all SBValues are representable as Int's or Floats, Bool's etc.  But I don't see any way to report errors with these new API's.

Reply:
if these errors matter to users then `llvm::Expected<llvm::APSInt>` would be the right choice. Otherwise std::optional is the way to go.